modules/signals/src/patch-state.ts

```diff
@@ -6,7 +6,9 @@ export type PartialStateUpdater<State extends object> = (
 
 export function patchState<State extends object>(
   stateSignal: StateSignal<State>,
-  ...updaters: Array<Partial<State & {}> | PartialStateUpdater<State & {}>>
+  ...updaters: NoInfer<
```

Comment:
NoInfer to resolve #4279
We already have a test case for this at https://github.com/ngrx/platform/blob/bc49af13bcef576f499dbadd4466ea65469fbac1/modules/signals/entities/spec/with-entities.spec.ts#L74, which fails without this fix.

This is a breaking change for the users that aren't on TypeScript 5.4.x